### DISCUSSION TITLE
Concerns about an Warning Message with groundwater signatures module

### DISCUSSION THREAD

**Discussion by mathiastidjani** [pastas/pastas#747] [Q&A] [answered]
I have a question regarding the groundwater signatures module. Is it fully operational? 

While using it, I get this error message that concerns me (WARNING: The estimated recession constant (1000.00) is close to the boundary. This may lead to incorrect results)

![Error_Pastas](https://github.com/pastas/pastas/assets/132659489/c1440981-6e96-4e1e-a597-c5a2edc3df7d)

**Comment by dbrakenhoff** [comment, marked as answer]
Hi @mathiastidjani,

The module is operational, but one of the methods is generating a warning. If you look at the code for the recovery constant, you will see that it is attempting to fit an exponential function, and the parameters are bounded. The upper bound is 1000. If the parameter reaches the bound in the optimization, the result might not be correct. In that situation it might worthwhile to explore for yourself whether the result you're seeing is reasonable or not. 

https://github.com/pastas/pastas/blob/eeb8e2b33783ddb2031112625b4552a4c3232428/pastas/stats/signatures.py#L1288-L1356

Hope that helps somewhat!

**Comment by mathiastidjani** [comment]
Dear David,

Thank you for your reply.

Is it possible to have references regarding thresholds or interpretation intervals for the various signatures?

Best regards
________________________________
De : Davíd Brakenhoff ***@***.***>
Envoyé : lundi 22 avril 2024 14:54
À : pastas/pastas ***@***.***>
Cc : Akominon Tidjani ***@***.***>; Mention ***@***.***>
Objet : Re: [pastas/pastas] Concerns about an Error Message with groundwater signatures module (Discussion #747)

Vous ne recevez pas souvent de courriers de la part de ***@***.*** Découvrez pourquoi cela est important<https://aka.ms/LearnAboutSenderIdentification>

Hi @mathiastidjani<https://github.com/mathiastidjani>,

The module is operational, but one of the methods is generating a warning. If you look at the code for the recovery constant, you will see that it is attempting to fit an exponential function, and the parameters are bounded. The upper bound is 1000. If the parameter reaches the bound in the optimization, the result might not be correct. In that situation it might worthwhile to explore for yourself whether the result you're seeing is reasonable or not.

https://github.com/pastas/pastas/blob/eeb8e2b33783ddb2031112625b4552a4c3232428/pastas/stats/signatures.py#L1288-L1356

Hope that helps somewhat!

**Reply by dbrakenhoff** [reply]
Check the references section in this notebook in the examples section of the documentation: https://pastas.readthedocs.io/latest/examples/signatures.html

Or check the docstrings of the signature functions themselves, these also contain references to literature where appropriate.
